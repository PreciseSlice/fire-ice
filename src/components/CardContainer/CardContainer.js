import React from 'react';
//import PropTypes from 'prop-types'
import { connect } from 'react-redux';
import Card from '../Card/Card' 
//import './CardContainer.css';

export const CardContainer = ({ houseData }) => {
  
  if(houseData) {
    const cards = houseData.map( house => {      
      return <Card data={house} key={house.name} />
    }) 
    
    return <div className="card-container">{cards}</div>
  } else {
    return null
  }
}

export const mapStateToProps = state => ({
  houseData: state.houseData,
});

export default connect(mapStateToProps, null)(CardContainer);
