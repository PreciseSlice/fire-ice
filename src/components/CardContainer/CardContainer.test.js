/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { CardContainer } from './CardContainer';

describe('CardContainer', () => {
  let wrapper;

  const mockData = [
    {
      house: 'stark'
    },
    {
      house: 'lannister'
    }
  ];

  beforeEach(() => {
    wrapper = shallow(<CardContainer houseData={mockData} />);
  });

  it('exist and matches snapshot', () => {
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
});
