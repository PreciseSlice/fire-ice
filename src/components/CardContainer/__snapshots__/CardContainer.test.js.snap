// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer exist and matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    houseData={
        Array [
            Object {
              "house": "stark",
            },
            Object {
              "house": "lannister",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          data={
                    Object {
                              "house": "stark",
                            }
          }
/>,
        <Unknown
          data={
                    Object {
                              "house": "lannister",
                            }
          }
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "house": "stark",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "house": "lannister",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            data={
                        Object {
                                    "house": "stark",
                                  }
            }
/>,
          <Unknown
            data={
                        Object {
                                    "house": "lannister",
                                  }
            }
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "house": "stark",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "house": "lannister",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
